# Description: Don't assume that arm_version>=7 means that neon code
#  should be built. Instead, use arm_neon==1 or arm_neon_optional==1
# Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/third_party/webrtc/common_audio/common_audio.gyp b/third_party/webrtc/common_audio/common_audio.gyp
--- a/third_party/webrtc/common_audio/common_audio.gyp
+++ b/third_party/webrtc/common_audio/common_audio.gyp
@@ -141,17 +141,17 @@
             'signal_processing/complex_bit_reverse_arm.S',
             'signal_processing/spl_sqrt_floor_arm.S',
           ],
           'sources!': [
             'signal_processing/complex_bit_reverse.c',
             'signal_processing/spl_sqrt_floor.c',
           ],
           'conditions': [
-            ['arm_version>=7', {
+            ['arm_neon==1 or arm_neon_optional==1', {
               'dependencies': ['common_audio_neon',],
               'sources': [
                 'signal_processing/filter_ar_fast_q12_armv7.S',
               ],
               'sources!': [
                 'signal_processing/filter_ar_fast_q12.c',
               ],
             }],
@@ -207,17 +207,17 @@
               'xcode_settings': {
                 'OTHER_CFLAGS': [ '-msse2', ],
               },
             }],
           ],
         },
       ],  # targets
     }],
-    ['target_arch=="arm" and arm_version>=7 or target_arch=="arm64"', {
+    ['target_arch=="arm" and (arm_neon==1 or arm_neon_optional==1) or target_arch=="arm64"', {
       'targets': [
         {
           'target_name': 'common_audio_neon',
           'type': 'static_library',
           'includes': ['../build/arm_neon.gypi',],
           'sources': [
             'fir_filter_neon.cc',
             'resampler/sinc_resampler_neon.cc',
